apps/opencs/view/render/terrainselection.cpp

```diff
@@ -0,0 +1,251 @@
+    const auto y = static_cast<int>(std::floor(yd));
+
+    return std::make_pair(x, y);
+}
+
+std::pair<int, int> CSVRender::TerrainSelection::toVertexCoords(osg::Vec3d worldPos) const
+{
+    const double xd {worldPos.x() * landTextureSize / cellSize+(landTextureSize / cellSize)/4};
+    const double yd {worldPos.y() * landSize / cellSize+(landTextureSize / cellSize)/4};
+
+    const auto x = static_cast<int>(std::floor(xd));
+    const auto y = static_cast<int>(std::floor(yd));
+
+    return std::make_pair(x, y);
+}
+
+double CSVRender::TerrainSelection::toWorldCoords(int pos)
+{
+    return cellSize * static_cast<double>(pos) / landTextureSize;
+}
+
+size_t CSVRender::TerrainSelection::landIndex(int x, int y)
+{
+    return y * landSize + x;
+}
+
+void CSVRender::TerrainSelection::update()
+{
+    osg::ref_ptr<osg::Geometry> newGeometry = new osg::Geometry();
+
+    // Color
+    osg::ref_ptr<osg::Vec4Array> colors = new osg::Vec4Array{};
+    colors->push_back(osg::Vec4f(0.f, 0.5f, 0.f, 1.f));
+
+    newGeometry->setColorArray(colors, osg::Array::Binding::BIND_OVERALL);
+
+    newGeometry->getOrCreateStateSet()->setMode(GL_LIGHTING, osg::StateAttribute::Values::OFF);
+
+    const osg::ref_ptr<osg::Vec3Array> vertices (new osg::Vec3Array);
+
+    if (!mSelection.empty())
+    {
+
+        for (std::pair<int, int> localPos : mSelection) {
+            int x {localPos.first};
+            int y {localPos.second};
+
+            // check adjacent terrain squares to see if there should be a border between them
+            const auto north = std::find(mSelection.begin(), mSelection.end(), std::make_pair(x, y + 1));
+            const auto south = std::find(mSelection.begin(), mSelection.end(), std::make_pair(x, y - 1));
+            const auto east = std::find(mSelection.begin(), mSelection.end(), std::make_pair(x + 1, y));
+            const auto west = std::find(mSelection.begin(), mSelection.end(), std::make_pair(x - 1, y));
+            const int textureSizeToLandSizeModifier = 4; // To-do: Calculate modifier according to landSize and landTextureSize
+
+            // Nudge selection by 1/4th of a texture size, similar how blendmaps are nudged
+            #define NUDGEPERCENTAGE (0.25)
+            const int nudgeOffset = (cellSize / landTextureSize) * NUDGEPERCENTAGE;
+            const int landHeightsNudge = (cellSize / landSize)/64; // Does this work with all land size configurations?
+
+            // calculate global vertex coordinates at selection box corners
+            int x1 = x * textureSizeToLandSizeModifier + landHeightsNudge;
+            int x2 = x * textureSizeToLandSizeModifier + textureSizeToLandSizeModifier + landHeightsNudge;
+            int y1 = y * textureSizeToLandSizeModifier - landHeightsNudge;
+            int y2 = y * textureSizeToLandSizeModifier + textureSizeToLandSizeModifier - landHeightsNudge;
+
+            // Draw straigth edges
+            int landHeightX1Y1 = calculateLandHeight(x1, y1),
```

Comment:
Please don't declare multiple variables that way. We have no shortage of ints. Might as well explicitly write it down for every variable.

Reply:
Do you mean that I should write "int" in front of each variabte, like.. int landHeightX1Y1; int ...X1Y2; and so on?

Reply:
Yes. Declaring multiple variables in a single statement is kinda frowned upon in modern C++.